packages/flutter/test/material/drawer_theme_test.dart

```diff
@@ -0,0 +1,162 @@
+        .toList();
+
+    expect(description, <String>[]);
+  });
+
+  testWidgets('Custom debugFillProperties', (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    DrawerThemeData(
+      backgroundColor: const Color(0x00000099),
+      scrimColor: const Color(0x00000098),
+      elevation: 5.0,
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(2.0)),
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[
+      'backgroundColor: Color(0x00000099)',
+      'scrimColor: Color(0x00000098)',
+      'elevation: 5.0',
+      'shape: RoundedRectangleBorder(BorderSide(Color(0xff000000), 0.0, BorderStyle.none), BorderRadius.circular(2.0))',
+    ]);
+  });
+
+  testWidgets('Default values are used when no Drawer or DrawerThemeData properties are specified', (WidgetTester tester) async {
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          key: scaffoldKey,
+          drawer: const Drawer(),
+        ),
+      ),
+    );
+    scaffoldKey.currentState!.openDrawer();
+    await tester.pumpAndSettle();
+
+    expect(_drawerMaterial(tester).color, null);
+    expect(_drawerMaterial(tester).elevation, 16.0);
+    expect(_drawerMaterial(tester).shape, null);
+    expect(_scrim(tester).color, Colors.black54);
+  });
+
+  testWidgets('DrawerThemeData values are used when no Drawer properties are specified', (WidgetTester tester) async {
+    const Color backgroundColor = Color(0x00000001);
+    const Color scrimColor = Color(0x00000002);
+    const double elevation = 7.0;
+    const RoundedRectangleBorder shape = RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0)));
+
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          drawerTheme: const DrawerThemeData(
+            backgroundColor: backgroundColor,
+            scrimColor: scrimColor,
+            elevation: elevation,
+            shape: shape,
+          ),
+        ),
+        home: Scaffold(
+          key: scaffoldKey,
+          drawer: const Drawer(),
+        ),
+      ),
+    );
+    scaffoldKey.currentState!.openDrawer();
+    await tester.pumpAndSettle();
+
+    expect(_drawerMaterial(tester).color, backgroundColor);
+    expect(_drawerMaterial(tester).elevation, elevation);
+    expect(_drawerMaterial(tester).shape, shape);
+    expect(_scrim(tester).color, scrimColor);
+  });
+
+  testWidgets('Drawer values take priority over DrawerThemeData values when both properties are specified', (WidgetTester tester) async {
```

Comment:
I think we're missing a similar test where `DrawerTheme` overrides `ThemeData.drawerTheme`

Reply:
Added